Compound.ipynb

```diff
@@ -468,6 +470,252 @@
+    "        img_json[\"pixel_size_x_unit\"] = str(px.getUnit())\n",
+    "        img_json[\"pixel_size_x_symbol\"] = px.getSymbol()\n",
+    "    if py is not None:\n",
+    "        img_json[\"pixel_size_y\"] = py.getValue()\n",
+    "    if image.getSizeT() > 1:\n",
+    "        img_json['deltaT'] = get_timestamps(conn, image)\n",
+    "    img_json['baseUrl'] = \"https://idr.openmicroscopy.org/webclient\"\n",
+    "    return img_json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_column(term, panel_x, panel_y):\n",
+    "    df1 = df[df['MoA'].str.contains(term, na=False, case=False)]\n",
+    "    uniques = df1[\"Compound\"].unique()\n",
+    "    for index, value in enumerate(uniques):\n",
+    "        df2 = df[df['Compound'].str.contains(value)]\n",
+    "        count = 0\n",
+    "        total = 0\n",
+    "        image_id = -1\n",
+    "        screen = \"\"\n",
+    "        for row in df2.index:\n",
+    "            concentration = df2['Concentration'][row]\n",
+    "            if concentration != \"unknown\":\n",
+    "                count = count + 1\n",
+    "                total = total + float(concentration)\n",
+    "                if image_id == -1:\n",
+    "                    image_id = df2['Image'][row]\n",
+    "        if image_id > -1:\n",
+    "            image = conn.getObject(\"Image\", image_id)\n",
+    "\n",
+    "            j = get_panel_json(image, panel_x, panel_y,\n",
+    "                               panel_width, panel_height)\n",
+    "            if index == 0:\n",
+    "                j['labels'].append(get_column_label(term))\n",
+    "            j['labels'].extend(get_image_label(total/count, value))\n",
+    "            panels_json.append(j)\n",
+    "            panel_y += panel_height + spacing\n",
+    "            panels_json[-1]['scalebar'] = get_scalebar_json()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create the figure "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "figure_json = {\"version\": 5}\n",
+    "panel_width = 200\n",
+    "panel_height = panel_width\n",
+    "spacing = panel_width/5\n",
+    "margin = 40\n",
+    "\n",
+    "panels_json = []\n",
+    "panel_x = margin\n",
+    "panel_y = margin\n",
+    "create_column(\"Inhibitor\", panel_x, panel_y)\n",
+    "panel_x += panel_width + spacing\n",
+    "panel_y = margin\n",
+    "create_column(\"Antagonist\", panel_x, panel_y)\n",
+    "figure_json['panels'] = panels_json\n",
+    "figure_json[\"figureName\"] = \"Compound Mechanism of Action\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Save the figure as a json file\n",
+    "You can then download the ``.json`` file or copy its content. In OMERO.figure use ``File>Import from JSON...``."
```

Comment:
```suggestion
    "You can then download the ``.json`` file, by clicking ``File > Open``, selecting it and cliking ``File > Download``. Alternatively, you can copy the content of the ``json`` file directly from this notebook UI. In OMERO.figure use ``File>Import from JSON...``."
    
   
```

Reply:
the workflow works as described, up to the point when I tried to save the figure on another OMERO.server. This failed. We should mention this fact, as the Save button is active (used the "workshop" server) in figure.